### DISCUSSION TITLE
Can't understand how to read file

### DISCUSSION THREAD

**Discussion by selfuryon** [cuelang/cue#886] [Q&A] [answered]
Hello!
I found struct `Read` in `file` package, but I can't understand how to use it

**test.txt**
```
File content
```

**main.cue**
```
package main

import (
    "tool/file"
)

let ca = file.Read & {
    filename: "test.txt"
    contents: string
}

test: {
    contents: ca.contents
}
```

**attempt:**
```bash
λ cue eval
test: {
    contents: string
}
```

**Comment by mpvl** [comment, marked as answer]
Basically, anything in the `tool` directory is intended to be interpreted by `cue cmd` (see `cue help cmd`). To normal CUE this is just data. These methods are designed to be "dynamic" and thus are not hermetic and not suitable for normal CUE evaluation.

Note, though, that we are considering allowing statically importing non-CUE files, either through the import mechanism or through a special embed form.

**Reply by selfuryon** [reply]
Thanks!

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/886.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
